iterating-over-data/practice-iterating-over-data.md

```diff
@@ -0,0 +1,465 @@
+
+
+##### !question
+
+Given this dictionary: 
+``` python
+
+my_profile = {"icecream_flavor": "chocolate", has_siblings: True}
+
+```
+
+Which loop(s) *can* be used to print out each key?
+
+##### !end-question
+
+##### !options
+
+* `for x in my_profile:`
+* `for x in my_profile.keys()`
+* `for x in my_profile.values()`
+* `for x, y in my_profile.items()`
+
+
+##### !end-options
+
+##### !answer
+
+* `for x in my_profile:`
+* `for x in my_profile.keys()`
+* `for x, y in my_profile.items()`
+
+##### !end-answer
+
+
+##### !hint
+
+##### !end-hint
+
+
+##### !rubric
+
+##### !end-rubric
+
+
+##### !explanation
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 4 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: checkbox
+* id: NyhIeJG0K
+* title: Choose the Best Option(s)
+
+
+##### !question
+
+Given this dictionary: 
+``` python
+
+my_profile = {"icecream_flavor": "chocolate", has_siblings: True}
+
+```
+
+Which loop(s) *can* be used to print out each value?
+
+##### !end-question
+
+##### !options
+
+* `for x in my_profile:`
+* `for x in my_profile.keys()`
+* `for x in my_profile.values()`
+* `for x, y in my_profile.items()`
+
+
+##### !end-options
+
+##### !answer
+
+* `for x in my_profile:`
+* `for x in my_profile.values()`
+* `for x, y in my_profile.items()`
+
+##### !end-answer
+
+
+##### !hint
+
+##### !end-hint
+
+
+##### !rubric
+
+##### !end-rubric
+
+
+##### !explanation
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 5 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: checkbox
+* id: V1VRgkzAY
+* title: Choose the Best Option(s)
+
+
+##### !question
+
+Given this dictionary: 
+``` python
+
+my_profile = {"icecream_flavor": "chocolate", has_siblings: True}
+
+```
+
+Which loop(s) *can* be used to print out both keys and values?
+
+##### !end-question
+
+##### !options
+
+* `for x in my_profile:`
+* `for x in my_profile.keys()`
+* `for x in my_profile.values()`
+* `for x, y in my_profile.items()`
+
+
+##### !end-options
+
+##### !answer
+
+* `for x in my_profile:`
+* `for x, y in my_profile.items()`
+
+##### !end-answer
+
+
+##### !hint
+
+##### !end-hint
+
+
+##### !rubric
+
+##### !end-rubric
+
+
+##### !explanation
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 6 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: code-snippet
+* language: python3.6
+* id: NkbCNAWCF
+* title: Match the Output
+
+##### !question
+
+Given a function named `search_dictionary` that takes two parameters: `dict` (a dictionary) and `check` (a string), check if the key `check` is found in `dict`, and then return the value at that key.
+
+For example, given: `{"icecream_flavor": "chocolate", "lucky_nums": [7, 18, 34], has_pets: True}` and `has_pets`, return `True`.
+
+##### !end-question
+
+##### !placeholder
+
+```python
+
+def search_dictionary(dict,check):
+  pass
```

Comment:
Same thing as before: Learn ends up highlighting `dict` so I recommend using a diff var name or adding this note.

Also, add a space between `dict,` and `check`

```suggestion
def search_dictionary(dict, check):
    # Note: the word "id" may be highlighted because the Learn platform believes it's reserved
    # This should not affect our code
    pass
```